File: mobile_apps/ALADDIN_iOS/Core/Network/APIService.swift
import Foundation
import Combine

/**
 * 🔌 API Service
 * Удобные методы для работы с API
 * Используется ViewModels для загрузки данных
 */

class APIService {
    
    private let networkManager: NetworkManager
    
    init(networkManager: NetworkManager) {
        self.networkManager = networkManager
    }
    
    // MARK: - VPN API
    
    func getVPNStatus(completion: @escaping (Result<VPNStatusResponse, Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.vpnStatus, completion: completion)
    }
    
    func connectVPN(completion: @escaping (Result<APIResponse<Bool>, Error>) -> Void) {
        struct EmptyBody: Codable {}
        networkManager.post(endpoint: AppConfig.Endpoint.vpnConnect, body: EmptyBody(), completion: completion)
    }
    
    func disconnectVPN(completion: @escaping (Result<APIResponse<Bool>, Error>) -> Void) {
        struct EmptyBody: Codable {}
        networkManager.post(endpoint: AppConfig.Endpoint.vpnDisconnect, body: EmptyBody(), completion: completion)
    }
    
    func getVPNServers(completion: @escaping (Result<[VPNServer], Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.vpnServers, completion: completion)
    }
    
    // MARK: - Family API
    
    func getFamilyMembers(completion: @escaping (Result<[FamilyMemberResponse], Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.familyMembers, completion: completion)
    }
    
    func addFamilyMember(name: String, role: String, completion: @escaping (Result<FamilyMemberResponse, Error>) -> Void) {
        struct AddMemberRequest: Codable {
            let name: String
            let role: String
        }
        networkManager.post(endpoint: AppConfig.Endpoint.addFamilyMember, body: AddMemberRequest(name: name, role: role), completion: completion)
    }
    
    func getFamilyStats(completion: @escaping (Result<FamilyStatsResponse, Error>) -> Void) {
        networkManager.get(endpoint: "/family/stats", completion: completion)
    }
    
    // MARK: - Analytics API
    
    func getAnalytics(period: String, completion: @escaping (Result<AnalyticsResponse, Error>) -> Void) {
        networkManager.get(endpoint: "\(AppConfig.Endpoint.analytics)?period=\(period)", completion: completion)
    }
    
    func getTopThreats(completion: @escaping (Result<[ThreatItem], Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.topThreats, completion: completion)
    }
    
    // MARK: - AI Assistant API
    
    func sendMessageToAI(message: String, completion: @escaping (Result<ChatMessageResponse, Error>) -> Void) {
        let request = ChatMessageRequest(
            message: message,
            userId: AppConfig.authToken ?? "guest",
            timestamp: Date()
        )
        networkManager.post(endpoint: AppConfig.Endpoint.aiSendMessage, body: request, completion: completion)
    }
    
    // MARK: - User API
    
    func getUserProfile(completion: @escaping (Result<UserProfile, Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.profile, completion: completion)
    }
    
    func updateProfile(name: String?, email: String?, phone: String?, completion: @escaping (Result<UserProfile, Error>) -> Void) {
        let request = UpdateProfileRequest(name: name, email: email, phone: phone)
        networkManager.post(endpoint: AppConfig.Endpoint.updateProfile, body: request, completion: completion)
    }
    
    // MARK: - Notifications API
    
    func getNotifications(completion: @escaping (Result<[NotificationResponse], Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.notifications, completion: completion)
    }
    
    func markNotificationAsRead(notificationId: String, completion: @escaping (Result<APIResponse<Bool>, Error>) -> Void) {
        struct MarkReadRequest: Codable {
            let notificationId: String
        }
        networkManager.post(endpoint: AppConfig.Endpoint.markRead, body: MarkReadRequest(notificationId: notificationId), completion: completion)
    }
    
    // MARK: - Subscription API
    
    func getTariffs(completion: @escaping (Result<[TariffResponse], Error>) -> Void) {
        networkManager.get(endpoint: AppConfig.Endpoint.tariffs, completion: completion)
    }
    
    func subscribe(tariffId: String, completion: @escaping (Result<SubscriptionStatus, Error>) -> Void) {
        struct SubscribeRequest: Codable {
            let tariffId: String
        }
        networkManager.post(endpoint: AppConfig.Endpoint.subscribe, body: SubscribeRequest(tariffId: tariffId), completion: completion)
    }
    
    // MARK: - Auth API
    
    func login(email: String, password: String, completion: @escaping (Result<LoginResponse, Error>) -> Void) {
        let request = LoginRequest(email: email, password: password)
        networkManager.post(endpoint: AppConfig.Endpoint.login, body: request, completion: completion)
    }
    
    func logout(completion: @escaping (Result<APIResponse<Bool>, Error>) -> Void) {
        struct EmptyBody: Codable {}
        networkManager.post(endpoint: AppConfig.Endpoint.logout, body: EmptyBody(), completion: completion)
    }
}

File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/network/ApiService.kt
package family.aladdin.android.network

import family.aladdin.android.models.*
import retrofit2.http.*

/**
 * 🔌 API Service
 * Retrofit интерфейс для всех API endpoints
 */

interface ApiService {
    
    // ═══════════════════════════════════════════════════════════════
    // VPN API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/vpn/status")
    suspend fun getVPNStatus(): VPNStatusResponse
    
    @POST("/api/vpn/connect")
    suspend fun connectVPN(): ApiResponse<Boolean>
    
    @POST("/api/vpn/disconnect")
    suspend fun disconnectVPN(): ApiResponse<Boolean>
    
    @GET("/api/vpn/servers")
    suspend fun getVPNServers(): List<VPNServer>
    
    // ═══════════════════════════════════════════════════════════════
    // Family API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/family/members")
    suspend fun getFamilyMembers(): List<FamilyMemberResponse>
    
    @POST("/api/family/add")
    suspend fun addFamilyMember(@Body request: AddMemberRequest): FamilyMemberResponse
    
    @GET("/api/family/stats")
    suspend fun getFamilyStats(): FamilyStatsResponse
    
    @DELETE("/api/family/remove/{memberId}")
    suspend fun removeFamilyMember(@Path("memberId") memberId: String): ApiResponse<Boolean>
    
    // ═══════════════════════════════════════════════════════════════
    // Analytics API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/analytics")
    suspend fun getAnalytics(@Query("period") period: String): AnalyticsResponse
    
    @GET("/api/analytics/top-threats")
    suspend fun getTopThreats(): List<ThreatItem>
    
    // ═══════════════════════════════════════════════════════════════
    // AI Assistant API
    // ═══════════════════════════════════════════════════════════════
    
    @POST("/api/ai/message")
    suspend fun sendMessageToAI(@Body request: ChatMessageRequest): ChatMessageResponse
    
    // ═══════════════════════════════════════════════════════════════
    // User API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/user/profile")
    suspend fun getUserProfile(): UserProfile
    
    @POST("/api/user/update")
    suspend fun updateProfile(@Body request: UpdateProfileRequest): UserProfile
    
    @POST("/api/user/password")
    suspend fun changePassword(@Body request: ChangePasswordRequest): ApiResponse<Boolean>
    
    // ═══════════════════════════════════════════════════════════════
    // Parental Control API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/parental/control/{childId}")
    suspend fun getParentalControlSettings(@Path("childId") childId: String): ParentalControlSettings
    
    @POST("/api/parental/limits")
    suspend fun updateParentalLimits(@Body request: ParentalControlSettings): ApiResponse<Boolean>
    
    @GET("/api/parental/child-stats/{childId}")
    suspend fun getChildStats(@Path("childId") childId: String): ChildStatsResponse
    
    // ═══════════════════════════════════════════════════════════════
    // Notifications API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/notifications")
    suspend fun getNotifications(): List<NotificationResponse>
    
    @POST("/api/notifications/read")
    suspend fun markNotificationAsRead(@Body request: MarkReadRequest): ApiResponse<Boolean>
    
    // ═══════════════════════════════════════════════════════════════
    // Subscription API
    // ═══════════════════════════════════════════════════════════════
    
    @GET("/api/subscription/tariffs")
    suspend fun getTariffs(): List<TariffResponse>
    
    @POST("/api/subscription/subscribe")
    suspend fun subscribe(@Body request: SubscribeRequest): SubscriptionStatus
    
    @POST("/api/subscription/cancel")
    suspend fun cancelSubscription(): ApiResponse<Boolean>
    
    // ═══════════════════════════════════════════════════════════════
    // Auth API
    // ═══════════════════════════════════════════════════════════════
    
    @POST("/api/auth/login")
    suspend fun login(@Body request: LoginRequest): LoginResponse
    
    @POST("/api/auth/logout")
    suspend fun logout(): ApiResponse<Boolean>
    
    @POST("/api/auth/register")
    suspend fun register(@Body request: RegisterRequest): LoginResponse
    
    // ═══════════════════════════════════════════════════════════════
    // QR Payment API (для России)
    // ═══════════════════════════════════════════════════════════════
    
    @POST("/api/payments/qr/create")
    suspend fun createQRPayment(
        @Header("Authorization") token: String,
        @Body request: Map<String, Any>
    ): family.aladdin.android.viewmodels.CreateQRPaymentResponse
    
    @GET("/api/payments/qr/status/{paymentId}")
    suspend fun checkQRPaymentStatus(
        @Header("Authorization") token: String,
        @Path("paymentId") paymentId: String
    ): family.aladdin.android.viewmodels.CheckQRPaymentStatusResponse
}

// ═══════════════════════════════════════════════════════════════
// Request Models
// ═══════════════════════════════════════════════════════════════

data class AddMemberRequest(
    val name: String,
    val role: String
)

data class ChangePasswordRequest(
    val oldPassword: String,
    val newPassword: String
)

data class MarkReadRequest(
    val notificationId: String
)

data class SubscribeRequest(
    val tariffId: String
)

